Makefile

```diff
@@ -207,30 +210,44 @@ e2e-helm-install:
 	cd .staging/helm && tar -xvf helmbin.tar.gz
 	./.staging/helm/linux-amd64/helm version --client
 
+e2e-docker-credential-store-setup:
+	rm -rf .staging/pass
+	mkdir -p .staging/pass
+	cd .staging/pass && git clone https://github.com/docker/docker-credential-helpers.git
+	cd .staging/pass/docker-credential-helpers && make pass
+	cp .staging/pass/docker-credential-helpers/bin/build/docker-credential-pass /usr/local/bin/
+	sed -i '0,/{/s/{/{\n\t"credsStore": "pass",/' ~/.docker/config.json
+
+	gpg --batch --passphrase '' --quick-gen-key ratify default default
+	pass init $$(gpg --list-keys ratify | sed -n 2p | tr -d " \t\n\r")
+
 e2e-notaryv2-setup:
 	rm -rf .staging/notaryv2
 	mkdir -p .staging/notaryv2
 	cd .staging/notaryv2 && git clone https://github.com/notaryproject/notation
 	cd .staging/notaryv2/notation && make build
 
 	echo 'FROM alpine\nCMD ["echo", "notaryv2 signed image"]' > .staging/notaryv2/Dockerfile
-	docker build -t ${LOCAL_TEST_REGISTRY}/notation:signed .staging/notaryv2
-	docker push ${LOCAL_TEST_REGISTRY}/notation:signed
-	echo 'FROM alpine\nCMD ["echo", "notaryv2 signed image auth enabled"]' > .staging/notaryv2/Dockerfile
-	docker build -t ${LOCAL_TEST_REGISTRY_AUTH}/notation:signed .staging/notaryv2
-	docker push ${LOCAL_TEST_REGISTRY_AUTH}/notation:signed
+	docker build --no-cache -t ${TEST_REGISTRY}/notation:signed .staging/notaryv2
+	docker push ${TEST_REGISTRY}/notation:signed
 
 	docker pull ${LOCAL_UNSIGNED_IMAGE}
-	docker image tag ${LOCAL_UNSIGNED_IMAGE} ${LOCAL_TEST_REGISTRY}/notation:unsigned
-	docker push ${LOCAL_TEST_REGISTRY}/notation:unsigned
-	docker image tag ${LOCAL_UNSIGNED_IMAGE} ${LOCAL_TEST_REGISTRY_AUTH}/notation:unsigned
-	docker push ${LOCAL_TEST_REGISTRY_AUTH}/notation:unsigned
+	docker image tag ${LOCAL_UNSIGNED_IMAGE} ${TEST_REGISTRY}/notation:unsigned
+	docker push ${TEST_REGISTRY}/notation:unsigned
 
 	rm -rf ~/.config/notation
 	.staging/notaryv2/notation/bin/notation cert generate-test --default "ratify-bats-test"
-	.staging/notaryv2/notation/bin/notation sign `docker image inspect ${LOCAL_TEST_REGISTRY}/notation:signed | jq -r .[0].RepoDigests[0]`
-	.staging/notaryv2/notation/bin/notation sign `docker image inspect ${LOCAL_TEST_REGISTRY}/all:v0 | jq -r .[0].RepoDigests[0]`
-	.staging/notaryv2/notation/bin/notation sign -u ${LOCAL_TEST_REGISTRY_USERNAME} -p ${LOCAL_TEST_REGISTRY_PASSWORD} `docker image inspect ${LOCAL_TEST_REGISTRY_AUTH}/notation:signed | jq -r .[0].RepoDigests[0]`
+	.staging/notaryv2/notation/bin/notation sign `docker image inspect ${TEST_REGISTRY}/notation:signed | jq -r .[0].RepoDigests[0]`
+	.staging/notaryv2/notation/bin/notation sign `docker image inspect ${TEST_REGISTRY}/all:v0 | jq -r .[0].RepoDigests[0]`
+
+	if [ "${USE_REMOTE_REGISTRY}" != "true" ]; then \
```

Comment:
Just a note that I will be removing the auth enabled vs not not auth enabled local registry in the Cosign refactor PR since Cosign not supports auth